### PR TITLE
Empty Response feedback for 204 on Lightspeed

### PR DESCRIPTION
**Pull request by hasys**
VS Code counterpart of https://github.com/ansible/ansible-ai-connect-service/pull/979

### CODE REVIEW

File: src/features/lightspeed/errors.ts [hunk 1]
```diff
@@ -117,34 +117,9 @@ export const ERRORS_CONNECTION_TIMEOUT = new Error(
...
-);
-ERRORS.addError(
-  204,
-  new Error(
-    "error__wca_empty_response",
-    "IBM watsonx Code Assistant returned an empty response. Please contact your administrator.",
-  ),
+export const ERRORS_EMPTY_RESPONSE = new Error(
+  "empty__response",
+  "Ansible Lightspeed didn't find suggestion for your request. Please try reword.",
```
Comment (manstis) [inline src/features/lightspeed/errors.ts:122]: Is this the message we used to show for `204`'s?

The four possible `204`'s we return represent different scenarios.

IDK if a more generic message should be used?

Comment (manstis) [reply src/features/lightspeed/errors.ts:122]: Fair enough.. @robinbobbitt is usually very good at suggesting wording.

I'm not one to get into length discussions about it.. and she usually changes what I propose anyway :smile_cat: 

I assume changing the 204's in the server side is still work to be completed, or is it [dying a death](https://en.wiktionary.org/wiki/die_a_death#:~:text=To%20fail%20or%20flop%3B%20to%20lose%20popularity.) now?

Comment (robinbobbitt) [reply src/features/lightspeed/errors.ts:122]: @hasys I was thinking something like "Ansible Lightspeed does not have a suggestion for this input. Try changing your prompt, or contact your administrator with Suggestion Id the_suggestion_id for assistance." wdyt?

@manstis my hope was we could revisit the 204 status code when we GA the API, i.e. v1.

**Review by goneri — APPROVED** [review]

**Comment by hasys** [comment]
Hello @manstis, @robinbobbitt, I updated the PR

now when server sends 204, user will see:
Screenshot 2024-05-30 at 13 14 28

For the record. Times to times (pretty often) instead of the screenshot above user will see:
Screenshot 2024-05-30 at 13 25 57

The error is happening here: https://github.com/ansible/vscode-ansible/blob/bf79bf62b0563928db7c5c00d119b7410dc6a82d/src/features/lightspeed/api.ts#L117

The request throws an `Error: Parse Error: Expected HTTP/` error and user see the general error after it. I have tried to understand what is going on there, but I think I already spent too much there and I am out of ideas.

Since I think it is not related to the issue I am working on I suggest to report it as separate issue and deal with it in separate PR.

P.S. There is a diff for two requests: successful and not successful:
```diff
kgaevski:~ % diff file1 file2
3,4c3,4
<                 b'pache2\\n"],"model":"mocked_wca_client","suggestionId":"d8259'
<                 b'b8f-402d-4ed9-8ffa-77dec8e67669"}'],
---
>                 b'pache2\\n"],"model":"mocked_wca_client","suggestionId":"10e93'
>                 b'2ab-d9af-4fa7-a18c-d23238a3cc34"}'],
12c12
<  'accepted_renderer': <rest_framework.renderers.JSONRenderer object at 0x13a6e6b10>,
---
>  'accepted_renderer': <rest_framework.renderers.JSONRenderer object at 0x13a6e7990>,
24c24
<                       'view': <ansible_ai_connect.ai.api.views.Completions object at 0x13a689f90>},
---
>                       'view': <ansible_ai_connect.ai.api.views.Completions object at 0x13a7e64d0>},
```

And the full response dump:
```js
{'_charset': None,
 '_container': [b'{"predictions":["      ansible.builtin.apt:\\n        name: a'
                b'pache2\\n"],"model":"mocked_wca_client","suggestionId":"10e93'
                b'2ab-d9af-4fa7-a18c-d23238a3cc34"}'],
 '_handler_class': None,
 '_is_rendered': True,
 '_post_render_callbacks': [],
 '_reason_phrase': None,
 '_request': None,
 '_resource_closers': [],
 'accepted_media_type': 'application/json',
 'accepted_renderer': <rest_framework.renderers.JSONRenderer object at 0x13a6e7990>,
 'closed': False,
 'content_type': None,
 'context_data': None,
 'cookies': <SimpleCookie: >,
 'data': None,
 'exception': False,
 'headers': {'Content-Type': 'None', 'Allow': 'POST, OPTIONS', 'Content-Security-Policy-Report-Only': "default-src 'self' data:; style-src 'self' 'unsafe-inline'", 'Content-Length': '0'},
 'renderer_context': {'args': (),
                      'kwargs': {},
                      'request': <rest_framework.request.Request: POST '/api/v0/ai/completions/'>,
                      'response': <Response status_code=204, "None">,
                      'view': <ansible_ai_connect.ai.api.views.Completions object at 0x13a7e64d0>},
 'status_code': 204,
 'tasks': [{'collection': 'ansible.builtin',
            'module': 'ansible.builtin.apt',
            'name': 'install go',
            'prediction': '      ansible.builtin.apt:\n'
                          '        name: apache2\n'}],
 'template_name': None,
 'using': None}
```

File: src/features/lightspeed/errors.ts (hunk 1 above)
Comment (hasys) [reply src/features/lightspeed/errors.ts]: Thank you @manstis, @robinbobbitt, fixed.

**Comment by hasys** [comment]
Created Jira issue for the remaining issue: https://issues.redhat.com/browse/AAP-24828

**Pull request merged**